addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3225,10 +3225,15 @@ export class OdooEditor extends EventTarget {
         const files = getImageFiles(ev.clipboardData);
         const targetSupportsHtmlContent = isHtmlContentSupported(sel.anchorNode);
         const clipboardHtml = ev.clipboardData.getData('text/html');
-        if (files.length && targetSupportsHtmlContent) {
-            this.addImagesFiles(files).then(html => this.execCommand('insertHTML', this._prepareClipboardData(html)));
-        } else if (clipboardHtml && targetSupportsHtmlContent) {
-            this.execCommand('insertHTML', this._prepareClipboardData(clipboardHtml));
+        const clipboardDiv = document.createElement('div');
+        clipboardDiv.innerHTML = clipboardHtml;
+        const clipboardHtmlIsImage = (clipboardDiv.firstChild instanceof HTMLImageElement);
+        if ((files.length || clipboardHtml) && targetSupportsHtmlContent) {
```

Comment:
Wait... If `!files.length`, then you're just basically calling `addImagesFiles([])`, which does nothing so your `if` case ends up doing exactly the same thing as your `else` case. What were you hoping to achieve here? And incidentally, what does it have to do with tables?